components/engine_traits/src/engine.rs

```diff
@@ -343,6 +346,10 @@ where
         PathBuf::from(&self.root_path)
     }
 
+    fn tablets_snap_path(&self) -> PathBuf {
+        PathBuf::from(&self.root_path).join("tablets_snap")
```

Comment:
I don't think this is something `TabletFactory` should care. Why not get the value from configuration directly?

Reply:
here config is `raftstore` config, do not contains KV storage data path. I think it should be initialize out of raftstore, I change it  with pass the param.